crates/rpc-types-engine/src/lib.rs

```diff
@@ -18,5 +18,154 @@
 
 extern crate alloc;
 
-mod payload;
-pub use payload::*;
+pub use alloy_rpc_types_engine::ExecutionPayloadV1;
+pub use op_alloy_genesis::{RollupConfig, SystemConfig};
+pub use op_alloy_protocol::L2BlockInfo;
+
+use alloy_eips::eip2718::Decodable2718;
+use alloy_primitives::B256;
+use op_alloy_consensus::{OpTxEnvelope, OpTxType};
+use op_alloy_protocol::{BlockInfo, L1BlockInfoBedrock, L1BlockInfoEcotone, L1BlockInfoTx};
+
+mod attributes;
+pub use attributes::{OptimismAttributesWithParent, OptimismPayloadAttributes};
+
+mod payload_v3;
+pub use payload_v3::OptimismExecutionPayloadEnvelopeV3;
+
+mod payload_v4;
+pub use payload_v4::OptimismExecutionPayloadEnvelopeV4;
+
+mod errors;
+pub use errors::{ToL2BlockRefError, ToSystemConfigError};
+
+/// Converts a payload into its most inner [ExecutionPayloadV1].
+pub trait IntoInnerPayload {
```

Comment:
this naming is a bit misleading, because this only borrows

Reply:
Agreed, fixing.